bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/handler/ZoneHandler.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.alarmdecoder.internal.handler;
+
+import static org.openhab.binding.alarmdecoder.internal.AlarmDecoderBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.alarmdecoder.internal.config.ZoneConfig;
+import org.openhab.binding.alarmdecoder.internal.protocol.ADMessage;
+import org.openhab.binding.alarmdecoder.internal.protocol.EXPMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ZoneHandler} is responsible for handling wired zones (i.e. REL & EXP messages).
+ *
+ * @author Bob Adair - Initial contribution
+ * @author Bill Forsyth - Initial contribution
+ */
+@NonNullByDefault
+public class ZoneHandler extends ADThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ZoneHandler.class);
+
+    private ZoneConfig config = new ZoneConfig();
+
+    public ZoneHandler(Thing thing) {
+        super(thing);
+    }
+
+    /** Construct zone id from address and channel */
+    public static final String zoneID(int address, int channel) {
+        return String.format("%d-%d", address, channel);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ZoneConfig.class);
+
+        if (config.address < 0 || config.channel < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid address/channel setting");
+            return;
+        }
+        logger.debug("Zone handler initializing for address {} channel {}", config.address, config.channel);
+
+        String id = zoneID(config.address, config.channel);
+        updateProperty(PROPERTY_ID, id); // set representation property used by discovery
+
+        initDeviceState();
+        logger.trace("Zone handler finished initializing");
+    }
+
+    @Override
+    protected void initDeviceState() {
```

Comment:
Almost all of the of the handlers implement this method the same way, is it possible to have it refactored?

Reply:
Yeah, I had actually meant to do that. Done.